src/core/geometry/qgsbox3d.cpp

```diff
@@ -133,3 +133,38 @@ bool QgsBox3d::operator==( const QgsBox3d &other ) const
          qgsDoubleNear( mZmin, other.mZmin ) &&
          qgsDoubleNear( mZmax, other.mZmax );
 }
+
+void QgsBox3d::scale( double scaleFactor, const QgsPoint *c )
```

Comment:
Can you add tests for theses methods? thanks

Reply:
done !